### PR TITLE
[oxigraph_server] Add dependency licenses

### PR DESCRIPTION
**Pull request by jeremiahpslewis**
This is an experiment for rust binaries so that we can package the full licenses of package dependencies. Not sure what the Yggdrasil policy is / what appropriate convention is as to what licenses need to come along into the jll, but cargo-license seems to offer an elegant solution and it is a practice followed (in some instances) by conda-forge. (see here: https://github.com/conda-forge/oxigraph-feedstock)

Ref: https://github.com/onur/cargo-license

### CODE REVIEW

File: O/oxigraph_server/build_tarballs.jl [hunk 1]
```diff
@@ -16,7 +16,10 @@ cd $WORKSPACE/srcdir/oxigraph/server
 
 cargo build --release --no-default-features --features=sled
 
-install_license $WORKSPACE/srcdir/oxigraph/LICENSE-MIT
+cargo install cargo-license
+cargo-license --avoid-build-deps --avoid-dev-deps --json > CARGO_LICENSES.json
```
Comment (giordano) [inline O/oxigraph_server/build_tarballs.jl:20]: You can't run programs built for the target.  Maybe this should built in a separate package and be a `HostBuildDependency`?

File: O/oxigraph_server/build_tarballs.jl [hunk 2]
```diff
@@ -36,6 +36,7 @@ products = Product[
 # Dependencies that must be installed before this package can be built
 dependencies = Dependency[
```
Comment (giordano) [inline O/oxigraph_server/build_tarballs.jl:37]:
```suggestion
dependencies = [
```

Comment (jeremiahpslewis) [reply O/oxigraph_server/build_tarballs.jl]: Gets me every time. Embarrassing. Thanks!

**Comment by giordano** [comment]
> Not sure what the Yggdrasil policy is / what appropriate convention is as to what licenses need to come along into the jll

Getting to the original question, we don't really have a policy for the case of static linking, simply because we almost never do it, apart from Rust and Go packages, which are a tiny minority of our ecosystem.

We want to ship in each tarball the license of the package itself, which is rather straightforward when we do dynamic linking: each tarball basically has a single package, and we have only its license.  However, with static linking you basically smash dozens of packages into a single build, so it isn't clear what license you want to ship: only the "final" package or all of its dependencies?  When in doubt, I usually look at Linux distributions.  Debian is a good one as they take the matter of license seriously (Arch Linux is less conscious of license), do they have Rust-based packages?  What they do in that case?

**Comment by jeremiahpslewis** [comment]
Fedora follows the following [practices](https://docs.fedoraproject.org/en-US/packaging-guidelines/Rust/#_license_for_binary_packages):

1. "the effective license for the subpackage containing the built binary must be calculated from the individual licenses of all dependencies"
2. "the list of individual licenses and the License tag with the effective license MUST be included"

On the other hand, Debian seems to just grab the top-level license and leave things at that, e.g. [ripgrep](https://metadata.ftp-master.debian.org/changelogs//main/r/rust-ripgrep/rust-ripgrep_13.0.0-2_copyright)

**Comment by giordano** [comment]
But as far as I can see [they only ship](https://fedora.pkgs.org/34/fedora-updates-x86_64/ripgrep-13.0.0-1.fc34.x86_64.rpm.html) the license files of ripgrep (which is exactly what we do as well), no?

**Comment by jeremiahpslewis** [comment]
Useful thread about this issue suggesting rust community still developing an approach: https://github.com/rust-lang/rust/issues/67014

**Pull request closed without merging**